### PR TITLE
task manager graceful shutdown support

### PR DESCRIPTION
**Pull request by optimistic9527**
https://github.com/beego/beego/issues/4631

### CODE REVIEW

File: task/task.go [hunk 1]
```diff
@@ -581,6 +591,17 @@ func (m *taskManager) StopTask() {
 	}()
 }
 
+// GracefulShutdown wait all task done
+func (m *taskManager) GracefulShutdown() <-chan struct{} {
```
Comment (flycash) [inline task/task.go:595]: How about we don't return the `chan`? I think `GracefulShutdown` means that we will block the users and then wait for the final result. So we just return nothing.

**Review by flycash — COMMENTED** [review]
Please add this PR into `CHANGELOG.md` and fix `golint` and `deepsource` checking. 

Thank you, nice job!

File: task/task.go (hunk 1 above)
Comment (optimistic9527) [reply task/task.go:595]:
```
   shutdown = m.GracefulShutdown()
	select {
	case <-shutdown:
		log.Printf("wait for done")
	case <-time.After(10 * time.Second):
		log.Printf("time out wait job")
	}
```
Worried about the task execution time being too long, the user can customize the timeout exit according to the situation,When all tasks are over, the channel will be closed,Don't worry about leaks.

**Comment by optimistic9527** [comment]
> Please add this PR into `CHANGELOG.md` and fix `golint` and `deepsource` checking.
> 
> Thank you, nice job!

I can’t fix the log DeepSource check

**Review by flycash — APPROVED** [review]
It seems that something wrong with Deepsource, let me fix it in the future.
And thank you very much.

**Pull request merged**